randutil/crypto_test.go

```diff
@@ -45,3 +48,79 @@ func (s *cryptoRandutilSuite) TestCryptoToken(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(b, HasLen, 5)
 }
+
+var (
+	KernelTestUUID        = "1031319a-b661-4c01-aafa-6def8a118944"
+	KernelUUIDPathDefault = randutil.KernelUUIDPath
+)
+
+func (s *cryptoRandutilSuite) TestRandomKernelUUIDNoFile(c *C) {
+	tmpDir := c.MkDir()
+	uuidPath := filepath.Join(tmpDir, "no-file")
+
+	randutil.KernelUUIDPath = uuidPath
+	defer func() {
+		randutil.KernelUUIDPath = KernelUUIDPathDefault
+	}()
+
+	value, err := randutil.RandomKernelUUID()
+	c.Check(value, Equals, "")
+	c.Check(err, ErrorMatches, "*.no such file or directory")
+}
+
+func (s *cryptoRandutilSuite) TestRandomKernelUUIDNoPerm(c *C) {
+	tmpDir := c.MkDir()
+	uuidPath := filepath.Join(tmpDir, "no-perm")
+	err := ioutil.WriteFile(uuidPath, []byte(KernelTestUUID), 0)
+	if err != nil {
+		c.Errorf("Write failed with: %s", err.Error())
+	}
+
+	randutil.KernelUUIDPath = uuidPath
+	defer func() {
+		randutil.KernelUUIDPath = KernelUUIDPathDefault
+	}()
+
+	value, err := randutil.RandomKernelUUID()
+	c.Check(value, Equals, "")
+	c.Assert(err, ErrorMatches, "*.permission denied")
+}
+
+func (s *cryptoRandutilSuite) TestRandomKernelUUID(c *C) {
+	for _, uuid := range []string{
+		KernelTestUUID,
+		" \t\n " + KernelTestUUID + " \n\t\r\n",
+	} {
+		// Create new path on each iteration because we cannot
+		// reuse previous path to read-only (0444) file.
+		tmpDir := c.MkDir()
```

Comment:
(nitpick^4) given that tmpDir is not used except in the next line you could just write: `uuidPath := filepath.Join(c.MkDir(), "uuid")`